app/models/response_node.rb

```diff
@@ -8,6 +8,10 @@ class ResponseNode < ApplicationRecord
   belongs_to :response
   has_closure_tree order: "new_rank", numeric_order: true, dont_order_roots: true, dependent: :destroy
 
+  before_save { children.each { |c| c.response_id = response_id } }
```

Comment:
Can we move this to a method? I prefer methods since it's a chance to give a nice juicy name to what you're doing.